### PR TITLE
Enable Prettier formatting for documentation

### PR DESCRIPTION
**Pull request by lb-**
- Exclude release notes as these are long lived and not currently aligned with Prettier preferences
- Exclude the .github/ folder for bespoke requirements in these template files
- Upgrade prettier version & re-format (one css file change)
- See https://github.com/wagtail/wagtail/discussions/8753
- Also https://github.com/wagtail/wagtail/pull/10579#discussion_r1239661644

### CODE REVIEW

**Comment by lb-** [comment]
Happy to add something that also reports the exact files that need changing and how, this is a long standing gap in Prettier but there is a work around. 

https://github.com/prettier/prettier/issues/6885#issuecomment-1473675037

I understand the main concern for this is that it adds friction for contributors, but think it's still worth doing.

File: .prettierignore [hunk 1]
```diff
@@ -8,7 +8,8 @@ coverage/
 static
 _build
 # File types which Prettier supports but we don’t want auto-formatting.
-*.md
+.github/
```
Comment (thibaudcolas) [inline .prettierignore:11]: I’m not sure about this one. Perhaps we should format it all by default and use `<!-- prettier-ignore -->` comments for parts of the files that have very specific formatting?

File: client/scss/components/_chooser.scss [hunk 2]
```diff
@@ -84,7 +84,14 @@
...
     // Uses a form-field class as well as widget classes, as all core and custom choosers use different widgets.
-    :is(.w-field--model_choice_field, .w-field--admin_task_chooser, .w-field--admin_page_chooser, .w-field--document_chooser_widget, .w-field--admin_image_chooser, .w-field--admin_snippet_chooser):is(:hover, :focus-within)
+    :is(
+        .w-field--model_choice_field,
+        .w-field--admin_task_chooser,
+        .w-field--admin_page_chooser,
+        .w-field--document_chooser_widget,
+        .w-field--admin_image_chooser,
+        .w-field--admin_snippet_chooser
+      ):is(:hover, :focus-within)
```
Comment (thibaudcolas) [inline client/scss/components/_chooser.scss:94]: Great!

File: .prettierignore (hunk 1 above)
Comment (laymonage) [reply .prettierignore:11]: It will be very weird to see that comment in the pull request and issue templates though

Comment (lb-) [reply .prettierignore:11]: I can just ignore the one specific file that we have added multiple line-breaks to

Comment (lb-) [reply .prettierignore:11]: I also tried the comment and it did not seem to work. 
Used syntax as per https://prettier.io/docs/en/ignore.html#markdown

**Comment by lb-** [comment]
Rebased this, only ignoring one file in the github folder now (Pull Request template) and also ignoring the contributors.md file.

I did some further investigation into a script to try a git diff output but I think this would not work in our CI.

You do get an output with at least the file name though, for example.

```
> prettier --check "**/?(.)*.{css,scss,js,ts,tsx,json,yaml,yml,md}"

Checking formatting...
[warn] docs/extending/index.md
[warn] Code style issues found in the above file(s). Forgot to run Prettier?
```

**Comment by lb-** [comment]
Going to mark this as a draft, may not be worth pursuing at this time but still keen on feedback from the Google Summer of Docs side of things - @activus-d

**Comment by lb-** [comment]
Closing this, maintaining this PR is a bit annoying. 

Still think it's worth a discussion but I'll bring this up at a core team meeting instead.

**Pull request closed without merging**